pkg/treemap/treemap.go

```diff
@@ -77,20 +77,43 @@ func (m *TreeMap[K, V]) Max() (key K, value V, ok bool) {
 	return key, value, false
 }
 
-// LowerBound returns an iterator pointing to the first element that is not less than key.
-// If no such element is found, a past-the-end iterator is returned.
-func (m *TreeMap[K, V]) LowerBound(key K) TreeMapIterator[K, V] {
-	iter := m.tree.Iterator()
-	iter.End()
+// Floor finds the floor key-value pair for the input key.
+// In case that no floor is found, then both returned values will be nil.
+// It's generally enough to check the first value (key) for nil, which determines if floor was found.
+//
```

Comment:
- Not `nil`, but is corresponding type's empty value.
- Not check key for nil, but check retunred `ok`.